pkg/syncutil/cachemanager/cachemanager.go

```diff
@@ -77,6 +244,115 @@ func (c *CacheManager) RemoveObject(ctx context.Context, instance *unstructured.
 	return nil
 }
 
+func (c *CacheManager) wipeData(ctx context.Context) error {
+	if _, err := c.opa.RemoveData(ctx, target.WipeData()); err != nil {
+		return err
+	}
+
+	// reset sync cache before sending the metric
+	c.syncMetricsCache.ResetCache()
+	c.syncMetricsCache.ReportSync()
+
+	return nil
+}
+
 func (c *CacheManager) ReportSyncMetrics() {
 	c.syncMetricsCache.ReportSync()
 }
+
+func (c *CacheManager) syncGVK(ctx context.Context, gvk schema.GroupVersionKind) error {
+	u := &unstructured.UnstructuredList{}
+	u.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   gvk.Group,
+		Version: gvk.Version,
+		Kind:    gvk.Kind + "List",
+	})
+
+	err := c.reader.List(ctx, u)
```

Comment:
We need to acquire a mutex here and check to see we are watching the GVK before we list it (we can release the mutex after listing).

If we don't acquire a mutex, then we have the possibility that a stale replay loop runs a `list` on a GVK we are no longer interested in. This would have the effect of re-establishing a watch on the GVK (something the k8s client does automatically).

Reply:
I added a locked call to List and check to see if the `gvk` is still in the watch set. Let me know if things don't look quite right please.